Mathlib/NumberTheory/NumberField/CanonicalEmbedding/FundamentalCone.lean

```diff
@@ -458,42 +460,141 @@ of the torsion of `K` is equal to the number of `integralPoint K` of norm `n`. -
+    x ∈ idealPoint K J ↔ x ∈ fundamentalCone K ∧ ∃ a : (𝓞 K), (a : 𝓞 K) ∈ (J : Set (𝓞 K)) ∧
+      mixedEmbedding K (a : 𝓞 K) = x := by
+  simp_rw [idealPoint, Set.mem_inter_iff, idealLattice, SetLike.mem_coe, FractionalIdeal.coe_mk0,
+    LinearMap.mem_range, LinearMap.coe_comp, LinearMap.coe_restrictScalars, coe_subtype,
+    Function.comp_apply, AlgHom.toLinearMap_apply, RingHom.toIntAlgHom_coe, Subtype.exists,
+    FractionalIdeal.mem_coe, FractionalIdeal.mem_coeIdeal, exists_prop', nonempty_prop,
+    exists_exists_and_eq_and]
+
+/-- The map that sends an `idealPoint` to an `integralPoint`. This map exists because `J` is an
+integral ideal. -/
+def idealPointMap : idealPoint K J → integralPoint K :=
+  fun ⟨a, ha⟩ ↦ ⟨a,
+    mem_integralPoint.mpr ⟨(mem_idealPoint.mp ha).1, (mem_idealPoint.mp ha).2.choose,
+      (mem_idealPoint.mp ha).2.choose_spec.2⟩⟩
+
+@[simp]
+theorem idealPointMap_apply (a : idealPoint K J) :
+    (idealPointMap K J a : mixedSpace K) = a := rfl
+
+theorem preimage_of_IdealPointMap (a : idealPoint K J) :
+    (preimageOfIntegralPoint (idealPointMap K J a) : 𝓞 K) ∈ (J : Set (𝓞 K)) := by
+  obtain ⟨_, ⟨x, hx₁, hx₂⟩⟩ := mem_idealPoint.mp a.prop
+  simp_rw [idealPointMap, ← hx₂, preimageOfIntegralPoint_mixedEmbedding]
+  exact hx₁
+
+/-- The map `idealPointMap` is actually an equiv between the `idealPoint K J` and the
+`integralPoint K` whose preimage lies in `J`. -/
+def idealPointEquiv : idealPoint K J ≃
+    {a : integralPoint K | (preimageOfIntegralPoint a : 𝓞 K) ∈ (J : Set (𝓞 K))} :=
+  Equiv.ofBijective (fun a ↦ ⟨idealPointMap K J a, preimage_of_IdealPointMap K J a⟩)
+    ⟨fun _ _ h ↦ (by
+        simp_rw [Subtype.ext_iff_val, idealPointMap_apply] at h
+        rwa [Subtype.ext_iff_val]),
+    fun ⟨a, ha₂⟩ ↦ ⟨⟨a.val,  mem_idealPoint.mpr ⟨a.prop.1,
+        ⟨preimageOfIntegralPoint a, ha₂, mixedEmbedding_preimageOfIntegralPoint a⟩⟩⟩, rfl⟩⟩
+
+variable {K J}
+
+theorem idealPointEquiv_apply (a : idealPoint K J) :
+    (idealPointEquiv K J a : mixedSpace K) = a := rfl
+
+theorem idealPointEquiv_symm_apply
+    (a : {a : integralPoint K // (preimageOfIntegralPoint a : 𝓞 K) ∈ (J : Set (𝓞 K)) }) :
+    ((idealPointEquiv K J).symm a : mixedSpace K) = a := by
+  rw [← (idealPointEquiv_apply ((idealPointEquiv K J).symm a)), Equiv.apply_symm_apply]
+
+theorem intNorm_idealPointEquiv_apply (a : idealPoint K J) :
+    intNorm (idealPointEquiv K J a).val = mixedEmbedding.norm (a : mixedSpace K) := by
+  rw [intNorm_coe, idealPointEquiv_apply]
+
+variable (K J)
+
+/-- For an integer `n`, The equivalence between the `idealPoint K` of norm `n` and the product
+of the set of nonzero principal ideals of `K` divisible by `J` of norm `n` and the
+torsion of `K`. -/
+def idealPointEquivNorm (n : ℕ) :
+    {a : idealPoint K J // mixedEmbedding.norm (a : mixedSpace K) = n} ≃
+      {I : (Ideal (𝓞 K))⁰ // (J : Ideal (𝓞 K)) ∣ I ∧ IsPrincipal (I : Ideal (𝓞 K)) ∧
+        absNorm (I : Ideal (𝓞 K)) = n} × (torsion K) := by
+  calc
```

Comment:
```suggestion
        absNorm (I : Ideal (𝓞 K)) = n} × (torsion K) := calc
    _ ≃ {a : {a : integralPoint K // (preimageOfIntegralPoint a).1 ∈ J.1} //
```
I *think* this is the suggested style, but the `by` is surely not needed.

Reply:
Yes, the `by` is not necessary. 

For the `calc` block, I try to look on the `Library Style Guidelines` page and there is says: 
```
As with by, the calc keyword should be placed on the line prior to the start of the calculation, with the calculation indented.
```
Although I do remember a discussion on Zulip where the style you suggested was mentionned...

Reply:
Well, if it's written let's follow the indication.